domain/controller/state/state.go

```diff
@@ -56,3 +56,46 @@ FROM   controller
 
 	return model.UUID(uuid.UUID), nil
 }
+
+// GetModelActivationStatus returns the activation status of a model.
+func (st *State) GetModelActivationStatus(ctx context.Context, controllerUUID string) (bool, error) {
+	db, err := st.DB()
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
+	type controllerModel struct {
```

Comment:
You don't need all of this information out of the database. You just need a bool. Let me know if you would like a hand to write this?

Reply:
Yup, I added it for redundancy to represent the model, will refactor it thanks tom :)